### PR TITLE
[Dynamo] Preserve source_fn in node.meta

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #92399

Sample value from the test case `test_export_with_stack_trace`

node.target | node.meta["source_fn"]
-- | --
aten.randn.default | <built-in method randn of type object at 0x7f8683263108>
aten.t.default | < built-in function linear >
aten.mm.default | < built-in function linear >
aten.cos.default | <built-in method cos of type object at 0x7f8683263108>
aten.relu.default | relu
aten.add.Tensor | < built-in function add >

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: test/dynamo/test_export.py [hunk 1]
```diff
@@ -894,6 +895,7 @@ def forward(self, x):
             if node.op not in {"placeholder", "output"}:
                 self.assertTrue(node.stack_trace is not None)
                 self.assertTrue(node.meta["nn_module_stack"] is not None)
+                self.assertTrue(node.meta["torch_api_call"] is not None)
```
Comment (jerryzh168) [inline test/dynamo/test_export.py:898]: should we test what it is as well? same for "nn_module_stack"

**Review by jerryzh168 — APPROVED** [review]
thanks a lot!

File: torch/_dynamo/output_graph.py [hunk 2]
```diff
@@ -778,6 +778,10 @@ def create_proxy(
         if nn_module_stack:
             rv.node.meta["nn_module_stack"] = nn_module_stack.copy()
 
+        torch_api_call = tx.torch_api_call
+        if torch_api_call:
+            rv.node.meta["torch_api_call"] = torch_api_call
```
Comment (SherlockNoMad) [inline torch/_dynamo/output_graph.py:783]: Need a better name for this. I don't want this only to be used for torch.* function. I suspect we will need to extend this for other buildin.functions and tensor.methods(). 

alternative names: original_fn, source_fn

Comment (jerryzh168) [reply torch/_dynamo/output_graph.py:783]: makes sense, we want to have things like `operator.add` here as well

Comment (SherlockNoMad) [reply torch/_dynamo/output_graph.py:783]: Update to source_fn.

Comment (jerryzh168) [reply torch/_dynamo/output_graph.py]: btw is `operator.add` supported now?

Comment (SherlockNoMad) [reply torch/_dynamo/output_graph.py]: I simplified the implementation. We can directly store the original node.target in node.meta["source_fn"], and propagate it along lowering. 

See updated summary. 

This now also cover the case for operator.add.

**Review by yanboliang — APPROVED** [review]

**Pull request closed without merging**